### PR TITLE
Fix compile catalog

### PR DESCRIPTION
**Pull request by sbrunner**
With one un existing or empty file, and an other

[https://github.com/camptocamp/c2cgeoportal/issues/1937]

### CODE REVIEW

File: buildtools/compile-catalog.js [hunk 1]
```diff
@@ -8,30 +6,25 @@ function main(inputs) {
...
-              cb(err);
-            });
-          } else {
-            process.stdout.write("{}");
+        fs.readFile(input, {encoding: 'utf-8'}, function(err, content) {
+          if (!err) {
+            contents.push(content);
+            if (contents.length == inputs.length) {
+              process.stdout.write(compiler.convertPo(contents.filter(function (content) {
+                return content.length != 0;
```
Comment (ger-benjamin) [inline buildtools/compile-catalog.js]: !== ?

File: buildtools/compile-catalog.js [hunk 2]
```diff
@@ -8,30 +6,25 @@ function main(inputs) {
...
-                contents.push(content);
-              }
-              cb(err);
-            });
-          } else {
-            process.stdout.write("{}");
+        fs.readFile(input, {encoding: 'utf-8'}, function(err, content) {
+          if (!err) {
+            contents.push(content);
+            if (contents.length == inputs.length) {
```
Comment (ger-benjamin) [inline buildtools/compile-catalog.js]: === ?

**Comment by ger-benjamin** [comment]
Two comments but looks good.

File: buildtools/compile-catalog.js (hunk 2 above)
Comment (tsauerwein) [inline buildtools/compile-catalog.js]: `contents.length` is always a number so there is no need to use the type-safe equals.

**Comment by sbrunner** [comment]
OK, thanks :-)

**Comment by ger-benjamin** [comment]
LGTM

**Pull request merged**